File: backend/src/services/healthProfileService.ts
import { HealthProfile } from "../models/healthProfile";
import { HealthProfileRepository } from "../repositories/healthProfileRepository";

export class HealthProfileService {
  constructor(private readonly repository = new HealthProfileRepository()) {}

  async getHealthProfile(id: number): Promise<HealthProfile | null> {
    return this.repository.findById(id);
  }

  async createHealthProfile(profile: HealthProfile): Promise<HealthProfile> {
    return this.repository.create(profile);
  }
}

File: mediremind_mobile/lib/services/health_profile_service.dart
import '../data/repositories/health_profile_repository.dart';
import '../models/health_profile.dart';

class HealthProfileService {
  HealthProfileService({HealthProfileRepository? repository})
    : _repository = repository ?? HealthProfileRepository();

  final HealthProfileRepository _repository;

  Future<HealthProfile> getProfile(String id) {
    return _repository.getProfile(id);
  }

  Future<HealthProfile> createProfile(HealthProfile profile) {
    return _repository.saveProfile(profile);
  }
}
